src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -452,7 +433,113 @@ Namespace Microsoft.CodeAnalysis.Operations
                 eventReference, Create(statement.Handler), adds:=adds, semanticModel:=_semanticModel, syntax:=statement.Syntax, type:=Nothing, constantValue:=Nothing, isImplicit:=True)
         End Function
 
-        Private Shared Function IsDelegateCreation(conversionKind As ConversionKind, conversionSyntax As SyntaxNode, operand As BoundNode, targetType As TypeSymbol) As Boolean
+#Region "Conversions"
+
+        ''' <summary>
+        ''' Creates the Lazy IOperation from a delegate creation operand or a bound conversion operand, handling when the conversion
+        ''' is actually a delegate creation.
+        ''' </summary>
+        Private Function GetConversionInfo(boundConversion As BoundConversionOrCast
+                                          ) As (Operation As Lazy(Of IOperation), Conversion As Conversion, IsDelegateCreation As Boolean)
+            Dim conversion = CreateConversion(boundConversion)
+            Dim boundOperand = GetConversionOperand(boundConversion)
+            If conversion.IsIdentity AndAlso boundConversion.ExplicitCastInCode Then
+                Dim adjustedInfo = TryGetAdjustedConversionInfo(boundConversion, boundOperand)
+
+                If adjustedInfo.Operation IsNot Nothing Then
+                    Return (Operation:=New Lazy(Of IOperation)(Function() adjustedInfo.Operation),
+                            adjustedInfo.Conversion,
+                            adjustedInfo.IsDelegateCreation)
+                End If
+            End If
+
+            Return (New Lazy(Of IOperation)(Function() CreateConversionOperand(boundOperand)),
```

Comment:
>CreateConversionOperand(boundOperand) [](start = 55, length = 37)

It looks like implementation of CreateConversionOperand called here  makes an assumption about what value is going to be returned for BoundDelegateCreationExpression by the IsDelegateCreation call below. I believe that the code would be more robust and easier to follow, if the assumptions were explicitly verified.

Reply:
Broke up the assumptions here.